addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,256 @@
+
+        int pollingInterval = getPollingInterval(appliance);
+        scheduler.schedule(this::updateChannels, pollingInterval, TimeUnit.SECONDS);
+    }
+
+    private int getPollingInterval(Appliance appliance) {
+        if (config.pollingInterval > 0) {
+            return config.pollingInterval;
+        }
+        return appliance.getConfig().getMeasurementTransmissionIntervall();
+    }
+
+    private void updateChannel(ChannelUID channel, Appliance appliance, Measurement measurement) {
+        String channelName = channel.getIdWithoutGroup();
+        State newState;
+        switch (channelName) {
+            case CHANNEL_NAME:
+                newState = new StringType(appliance.getName());
+                break;
+            case CHANNEL_PRESSURE:
+                newState = new DecimalType(measurement.getPressure());
+                break;
+            case CHANNEL_TEMPERATURE:
+                newState = new QuantityType<>(measurement.getTemperatureGuard(), SIUnits.CELSIUS);
+                break;
+            case CHANNEL_VALVE_OPEN:
+                newState = getValveOpenType(appliance);
+                break;
+            default:
+                throw new IllegalArgumentException("Channel " + channel + " not supported.");
+        }
+        updateState(channel, newState);
+    }
+
+    private OnOffType getValveOpenType(Appliance appliance) {
+        OndusService service = getOndusService();
+        if (service == null) {
+            return OnOffType.ON;
+        }
+        Optional<ApplianceCommand> commandOptional;
+        try {
+            commandOptional = service.getApplianceCommand(appliance);
+        } catch (IOException e) {
+            logger.debug("Could not get appliance command", e);
+            return OnOffType.ON;
+        }
+        if (!commandOptional.isPresent()) {
+            return OnOffType.ON;
+        }
+        return commandOptional.get().getCommand().getValveOpen() ? OnOffType.ON : OnOffType.OFF;
+    }
+
+    private Measurement getLastMeasurement(Appliance appliance) {
+        if (getOndusService() == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return new Measurement();
+        }
+
+        ApplianceData applianceData = getApplianceData(appliance);
+        if (applianceData == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Could not load data from API.");
+            return new Measurement();
+        }
+        List<Measurement> measurementList = applianceData.getData().getMeasurement();
+
+        if (measurementList.size() == 0) {
+            return new Measurement();
+        }
+
+        Measurement lastMeasurement = measurementList.get(measurementList.size() - 1);
+        return lastMeasurement;
```

Comment:
You don't need to allocate `lastMeasurement` since it is not used. 

Although not strictly necessary, I would personally make it more clear how the execution paths relate to each other. E.g. using:

```java
return measurementList.size() == 0 ? new Measurement() : measurementList.get(measurementList.size() - 1);
```

Reply:
You're right! Done :)